app/views/tag/_tagging.html.erb

```diff
@@ -1,9 +1,13 @@
 <!-- This is the sidebar tagging display, also renders Subscribe button for multiple subscription -->
 <div>
   <h1 style="display: inline-block;">Tags</h1>
-  <% if @node %>
-    <%= render partial: 'tag/subscribe_button', locals:{tags: @node.node_tags} %>
-  <% else %>
+  <% if @node && !@node.node_tags.empty? %>
+    <% if current_user && @node.liked_by(current_user.uid) %>
+      <%= render partial: 'tag/subscribe_button', locals:{tags: @node.node_tags} %>
+    <% elsif !current_user %>
+      <%= render partial: 'tag/subscribe_button', locals:{tags: @node.node_tags} %>
+    <% end %>
+  <% elsif @user && !@user.user_tags.empty? %>
```

Comment:
That's checking if list of tags is not empty, otherwise don't display the button.

Reply:
I want to know you are passing `locals:{tags: @user.user_tags}` if no node.tags exists. This is unclear to me.

Reply:
if if @node fails in the first condition(if @node is the very 1st check), then it would mean that no node tags exist and it would come to the next condition

Reply:
I am asking you why are you passing @user.user_tags if no nodetags exists as per one condition. Is it understandable now?
Sorry for misunderstanding

Reply:
Actually I am unable to go to a page that doesn't have any tags or delete the ones which have (accidentally blocked user) so I am not able to test this. Kindly suggest me a page which could have no tags whatsoever, I am so sorry to trouble you like this

Reply:
https://github.com/publiclab/plots2/pull/4555/files#diff-21c596e0afdbbb7a1a6c95f3ad6c9d87R7 this line is entered by you. I don't know why we need this.

Reply:
There is no line number 22 what are you saying?

Please remove it as it is not needed.

Reply:
I was talking about these lines sir, they were hidden:
<% else %>
  <span id="tags">
  <%= render partial: 'tag/tags', locals: { power_tag: false, label_name: 'label-primary', tags: @user.user_tags } %>
  </span>
<% end %>

I will remove it right away.

Reply:
On removing that elsif condtion from this PR, I am getting this, that is the button isn't coming:

![image](https://user-images.githubusercontent.com/40794215/51209808-24095d80-1937-11e9-9c4e-9fc4367cb257.png)

![image](https://user-images.githubusercontent.com/40794215/51209839-371c2d80-1937-11e9-89a8-98f46261afb0.png)

Reply:
Please let me know what is to be done. I am getting really tired right now, it's 2:35 am here and I have college in the morning. Let me know if you want this change. Thanks a lot.

Reply:
Yeah button should not come on the profile page. They are user defined tags. We don't want their subscriptions.
Button should appear on the map, question, wiki, and note. Check it

Reply:
Okay then, I'll commit this change

Reply:
Is the button coming on map, question, wiki and note? I am sure it will

Reply:
yes it is (on liking it like it's supposed to be)